### PR TITLE
infra: update the timeout of AFL

### PR DESCRIPTION
**Pull request by DavidKorczynski**
This solves timeouts for various of the existing fuzzers. It would be nice to have AFL run on these as the fuzzers are working. The following projects are examples of where the timeout triggers:
https://github.com/google/oss-fuzz/pull/5152/
https://github.com/google/oss-fuzz/pull/5164
https://github.com/google/oss-fuzz/pull/5150
https://github.com/google/oss-fuzz/pull/5148
https://github.com/google/oss-fuzz/pull/5147

### CODE REVIEW

**Comment by jonathanmetzman** [comment]
Not sure I like this. 1. We don't do the same in ClusterFuzz so this simply hides the fact that something is wrong. 2. Why 5 seconds? Why not 10? I think 1 second per input seems reasonable and i think it's fine if projects that can't meet this don't use AFL.

**Comment by inferno-chromium** [comment]
> Not sure I like this. 1. We don't do the same in ClusterFuzz so this simply hides the fact that something is wrong. 2. Why 5 seconds? Why not 10? I think 1 second per input seems reasonable and i think it's fine if projects that can't meet this don't use AFL.

I dont like the fact that we would miss on super nice bugs that afl++ can find (and libfuzzer) wont [@vanhauser-thc]. in cf, we just remove timeouts from corpus right, so in a way these targets do work?

**Comment by jonathanmetzman** [comment]
> > Not sure I like this. 1. We don't do the same in ClusterFuzz so this simply hides the fact that something is wrong. 2. Why 5 seconds? Why not 10? I think 1 second per input seems reasonable and i think it's fine if projects that can't meet this don't use AFL.
> 
> I dont like the fact that we would miss on super nice bugs that afl++ can find (and libfuzzer wont) [@vanhauser-thc]. in cf, we just remove timeouts from corpus right, so in a way these targets do work?

I didn't think of the fact we remove testcases from corpus. But I think there's still an edge case that raising the limit to 5 seconds misses. If every testcase takes longer than 1 second the fuzz target will pass bad build check but fail on CF.

**Comment by vanhauser-thc** [comment]
there is no upper limit on `-t` timeouts in afl and afl++. its just bad documentation :) 1000ms is just the default.
you can set -t 5000 for 5 seconds. 

There is another feature - adding a '+' sign to the timeout value. this will instruct afl-fuzz to skip timeout inputs.
so -t 5000+ should work fine - if there is at least one input that is below/equal 5000ms
(this is also present in vanilla afl, not a special afl++ feature).

the timeout value is something that is target dependent. setting this to a global "catch all" is not a good approach IMHO.

File: infra/base-images/base-runner/run_fuzzer [hunk 1]
```diff
@@ -114,6 +114,9 @@ if [[ "$FUZZING_ENGINE" = afl ]]; then
   test -e "$OUT/afl_cmplog.txt" && AFL_FUZZER_ARGS="$AFL_FUZZER_ARGS -l 2 -c $OUT/$FUZZER"
   # If $OUT/afl++.dict we load it as a dictionary for afl-fuzz.
   test -e "$OUT/afl++.dict" && AFL_FUZZER_ARGS="$AFL_FUZZER_ARGS -x $OUT/afl++.dict"
+  # Ensure timeout is a bit large than 1sec as some of the OSS-Fuzz fuzzers
+  # are slower than this. 
+  AFL_FUZZER_ARGS="$AFL_FUZZER_ARGS -t 50000"
```
Comment (vanhauser-thc) [inline infra/base-images/base-runner/run_fuzzer:119]: that would be 50 seconds - a tad too much?;)

**Comment by DavidKorczynski** [comment]
> I didn't think of the fact we remove testcases from corpus. But I think there's still an edge case that raising the limit to 5 seconds misses. If every testcase takes longer than 1 second the fuzz target will pass bad build check but fail on CF.

From my tests of the projects I highlight above the timeout occurs in the dry run - when this is passed the exec/sec increases a lot.

File: infra/base-images/base-runner/run_fuzzer (hunk 1 above)
Comment (DavidKorczynski) [reply infra/base-images/base-runner/run_fuzzer:119]: The thought process was in essence all or nothing, i.e. if the harness actually runs I thought it should get in and the plan was then to push the decision of timeouts over the Clusterfuzz due to the reasoning Abhishek highlight above.

Comment (vanhauser-thc) [reply infra/base-images/base-runner/run_fuzzer:119]: IMHO it makes no sense to have seeds that take longer than 5 seconds to complete. rather add the '+' so that afl-fuzz doesnt abort but skips longer running ones.
I think `-t 6000+` is maybe the best choice?

Comment (DavidKorczynski) [reply infra/base-images/base-runner/run_fuzzer:119]: Yeah I agree on that for sure. I have just tried simply adding `-t 1000+` and it seems to be working fairly well for the issue cases. Will update the PR in just a sec.

Comment (DavidKorczynski) [reply infra/base-images/base-runner/run_fuzzer]: I set it to `-t 5000+`

**Comment by DavidKorczynski** [comment]
@inferno-chromium @jonathanmetzman @vanhauser-thc switched it to `-t 5000+` so summary atm is cases that take longer than 5 sec will be discarded but AFL will continue to run regardless. This works well with the projects referenced above (at least the couple I tested).

**Comment by vanhauser-thc** [comment]
LGTM

**Comment by inferno-chromium** [comment]
We also autocalibrate to 1000+ when we see a lot of hangs in CF queue - https://github.com/google/clusterfuzz/blob/master/src/python/bot/fuzzers/afl/launcher.py#L920, maybe we need to change that 1000+ to 5000+, then it matches with what we have here. And some targets can be slow for 1 sec, so all testcases might time out and cause afl-fuzz to crash.
